### PR TITLE
Systemd fix

### PR DESCRIPTION
**Pull request by aleksandra-tarkowska**
few improvements based on work in devspace

https://github.com/openmicroscopy/devslave-c7-docker/issues/4

cc: @manics @joshmoore

### CODE REVIEW

**Comment by aleksandra-tarkowska** [comment]
I am not sure why because ENV is set https://github.com/aleksandra-tarkowska/ome-docker/blob/systemd_fix/omero-ssh-systemd/Dockerfile#L5, but following https://github.com/docker/docker/issues/18796#issuecomment-204462440 and adding 

```
    volumes:
        - /sys/fs/cgroup:/sys/fs/cgroup:ro
        - /run
     environment:
        - container=docker
```

it seems to work better

File: omero-ssh-systemd/Dockerfile [hunk 1]
```diff
@@ -14,10 +14,19 @@ RUN sed -i \
...
+RUN sed -i -e "s/Defaults    requiretty.*/ #Defaults    requiretty/g" /etc/sudoers
+
 RUN useradd omero && \
 	echo 'omero:omero' | chpasswd && \
 	echo "omero ALL= (ALL) NOPASSWD: ALL" >> /etc/sudoers.d/omero
 
+RUN rm -f /lib/systemd/system/systemd*udev* ; \
+	rm -f /lib/systemd/system/getty.target;
+
+RUN yum install -y initscripts \
```
Comment (manics) [inline omero-ssh-systemd/Dockerfile]: Why is this required?

Comment (aleksandra-tarkowska) [inline omero-ssh-systemd/Dockerfile]: to run `service` rather then `systemctl`, it makes all the devspace jobs and external scripts os independent, no?

**Comment by aleksandra-tarkowska** [comment]
regarding my comment https://github.com/ome/ome-docker/pull/36#issuecomment-210167269 that seems to help only with running single integration job, but if 2 run in the same time, still frozen.

File: omero-ssh-systemd/Dockerfile (hunk 1 above)
Comment (manics) [inline omero-ssh-systemd/Dockerfile]: True. On the other hand it means it's a slightly less accurate platform for testing since CentOS7 sysadmins won't have this package installed, and it means this image is becoming more tailored to our ci/test workflow instead of a general ssh/systemd container. If everyone else is happy with this then that's fine, but it'd be worth adding an explanation to the README along with a mention of the sudoers change.

Comment (jburel) [inline omero-ssh-systemd/Dockerfile]: I will try to keep the platform as accurate as possible
We can adjust the test for example as I did to use ```systemctl```
I would rather add another test if need be for other OS than losing accuracy

Comment (aleksandra-tarkowska) [inline omero-ssh-systemd/Dockerfile]: @jburel instead fo changing test script just add RUN yum install -y initscripts into Dockerfile

Comment (aleksandra-tarkowska) [inline omero-ssh-systemd/Dockerfile]: in https://github.com/ome/omero-install/blob/develop/linux/test/centos7_nginx/Dockerfile and https://github.com/ome/omero-install/blob/develop/linux/test/centos7_apache24/Dockerfile

Comment (aleksandra-tarkowska) [inline omero-ssh-systemd/Dockerfile]: https://github.com/ome/omero-install/blob/develop/linux/test/test_services.sh meant to be usable across other platforms, it cost nothing to just add `RUN yum install -y initscripts` into centos7 Dockerfiles

**Comment by aleksandra-tarkowska** [comment]
initscripts removed

File: omero-ssh-systemd/Dockerfile (hunk 1 above)
Comment (jburel) [inline omero-ssh-systemd/Dockerfile]: Better option

File: omero-ssh-systemd/Dockerfile [hunk 2]
```diff
@@ -14,10 +14,16 @@ RUN sed -i \
...
 	/etc/pam.d/sshd
 
+# To avoid error: sudo: sorry, you must have a tty to run sudo
+RUN sed -i -e "s/Defaults    requiretty.*/ #Defaults    requiretty/g" /etc/sudoers
+
 RUN useradd omero && \
 	echo 'omero:omero' | chpasswd && \
 	echo "omero ALL= (ALL) NOPASSWD: ALL" >> /etc/sudoers.d/omero
 
+RUN rm -f /lib/systemd/system/systemd*udev* ; \
```
Comment (manics) [inline omero-ssh-systemd/Dockerfile]: Can you add a link to https://bugzilla.redhat.com/show_bug.cgi?id=1046469#c11 so we can remove this workaround in future

**Comment by aleksandra-tarkowska** [comment]
done

**Comment by joshmoore** [comment]
@aleksandra-tarkowska : you'll re-list when ready?

Also, thoughts on extracting this so we can start giving it an official tag for re-use?

**Comment by aleksandra-tarkowska** [comment]
I think it is ready, relisted

**Comment by manics** [comment]
> Also, thoughts on extracting this so we can start giving it an official tag for re-use?

Is there a script (if not, want to write one since you're the git filter-branch guru)?

**Comment by aleksandra-tarkowska** [comment]
Could we tag that repo first please?
I am fixing devspace https://github.com/openmicroscopy/devspace/pull/37, will add tags

**Comment by joshmoore** [comment]
> Is there a script (if not, want to write one since you're the git filter-branch guru)?

Instructions are currently in https://github.com/openmicroscopy/ome-internal/issues/285 -- happy to process that 

> Could we tag that repo first please?

Which repo? This one? We can't, which is why I'd want to extract the dir.

**Comment by joshmoore** [comment]
Merging (after discussion re: tagging) so that I can extra it.

**Comment by joshmoore** [comment]
See https://github.com/openmicroscopy/omero-ssh-c7-docker/releases/tag/0.1.0 -- currently building on hub:

https://hub.docker.com/r/openmicroscopy/omero-ssh-c7/builds/btyt8jvr2fveobtkqvc6vqn/
